tools/rng_test/README.md

```diff
@@ -0,0 +1,151 @@
+# RNG(Random Number Generator) duplication testing tool
+
+This tool can be used to find duplicate random numbers generated by auth_generate_random() function.
+
+It repeats the following two operations.
+
+1. Call auth_generate_random() (executed by 'security_api' TASH command)
+2. Reboot
+
+
+
+### Prerequisites
+
+* Python 2.7 & 3.x
+
+* Install 'pySerial'
+
+  ```
+  ~$ pip install pyserial
+  ```
+
+  
+
+### How to use
+
+1. Enable Security API configurations in menuconfig.
+
+   * Application Configuration - Examples - Security Test 
+     * [*] Security API Test
+   * Security Framework 
+     * [*] Security API
+     * [*] Authentication
+     * [*] Cryptography
+     * [*] Key manager
+     * [*] Secure storage
+
+   
+
+2. Change settings if you need
+
+   * rng_test.py
+
+     ```python
+     # ------------- Global Variables -------------
+     port_num = '/dev/ttyUSB1'	# port number
+     baudrate = 115200
+     iterator = 1000				# Number of iteration you want
+     ...
+     ```
+
+     
+
+   * This script is made for the purpose of testing 16 bytes random values. If you want to test a larger random value, you need to modify 'PrintBuffer()' in 'security_api_utils.c' (to output random values in a single line).
+
+     ```c
+     // apps/examples/security_test/security_api/security_api_utils.c
+     void PrintBuffer(const char *header, unsigned char* buffer, uint32_t len)
+     {
+     	register uint32_t i = 0;
+     	printf("%s : %d\n", header, len);
+     	for (i = 0; i < len; i++) {
+     		if (i != 0 && i % 16 == 0) {
+     			printf("\n"); // comment out here
+     		}
+     		printf(" %02X", buffer[i]);
+     	}
+     	printf("\n");
+     }
+     ```
+
+     
+
+3. Run the script
+
+   ````
+   $ python rng_test.py
+   ````
+
+   * Result 
```

Comment:
Let's print result briefly first. I mean
how many times it runs and how many duplication is.

Reply:
I added codes that show total result. Thank you!